analysis/backtrace/backtrace.go

```diff
@@ -187,29 +195,48 @@ func (v *Visitor) Visit(s *df.AnalyzerState, entrypoint df.NodeWithTrace) {
 	// the analysis operates on data originating from every argument in every
 	// call to an entrypoint
 	for _, arg := range call.Args() {
-		v.visit(s, arg)
+		if err := v.visit(s, arg); err != nil {
+			v.Errs = append(v.Errs, err)
+		}
+
+		// report traces from entrypoint:
+		logger := s.Logger
+		traces := v.Traces[arg]
+		for i, trace := range traces {
+			logger.Infof("%v\n\n", trace)
+
+			// Stop if there is a limit on number of alarms (representing number of traces to report), and it has been reached.
+			if s.Config.MaxAlarms > 0 && i >= s.Config.MaxAlarms {
+				break
+			}
+
+			if err := validateTrace(trace); err != nil {
+				panic(fmt.Errorf("invalid trace: %v", err))
+			}
+		}
 	}
 }
 
 //gocyclo:ignore
-func (v *Visitor) visit(s *df.AnalyzerState, entrypoint *df.CallNodeArg) {
+func (v *Visitor) visit(s *df.AnalyzerState, entrypoint *df.CallNodeArg) (err error) {
+	logger := s.Logger
+
 	pos := entrypoint.Position(s)
 	if !pos.IsValid() {
 		pos = entrypoint.ParentNode().Position(s)
 	}
 
-	logger := s.Logger
-	logger.Infof("\n%s ENTRYPOINT %s", strings.Repeat("*", 30), strings.Repeat("*", 30))
-	logger.Infof("==> Node: %s\n", formatutil.Purple(entrypoint.String()))
-	logger.Infof("%s %s\n", formatutil.Green("Found at"), pos)
-
 	// Skip entrypoint if it is in a dependency or in the Go standard library/runtime
 	// TODO make this an option in the config
```

Comment:
There's a "context" field in the code identifiers that might be useful for that purpose.